tests/nosetests/pyanaconda_tests/module_payload_dnf_manager_test.py

```diff
@@ -163,3 +164,24 @@ def dump_configuration_test(self):
 
         msg = "installroot = /mnt/sysroot"
         self.assertTrue(any(map(lambda x: msg in x, cm.output)))
+
+    def get_installation_size_test(self):
+        """Test the get_installation_size method."""
+        # No transaction.
+        size = self.dnf_manager.get_installation_size()
+        self.assertEqual(size, Size("3000 MiB"))
+
+        # Fake transaction.
+        tsi_1 = Mock()
+        tsi_1.pkg.installsize = 1024 * 100
+        tsi_1.pkg.files = ["/file"] * 10
+
+        tsi_2 = Mock()
+        tsi_2.pkg.installsize = 1024 * 200
+        tsi_2.pkg.files = ["/file"] * 20
+
+        self.dnf_manager._base.transaction = [tsi_1, tsi_2]
```

Comment:
This code basically testing DNF. We don't want to test DNF code just that we are calling it correctly.

Reply:
Unfortunately, this is testing our code.

```
    def get_installation_size(self):
        """Calculate the installation size.

        :return: a space required by packages
        :rtype: an instance of Size
        """
        packages_size = Size(0)
        files_number = 0

        if self._base.transaction is None:
            return Size("3000 MiB")

        for tsi in self._base.transaction:
            # Space taken by all files installed by the packages.
            packages_size += tsi.pkg.installsize
            # Number of files installed on the system.
            files_number += len(tsi.pkg.files)

        log.debug("Space required for packages: %s", packages_size)

        # Calculate the files size depending on number of files.
        files_size = Size(files_number * DNF_EXTRA_SIZE_PER_FILE)
        log.debug("Space required for installed files: %s", files_size)

        # Get the total size. Add another 10% as safeguard.
        total_space = Size((packages_size + files_size) * 1.1)
        log.debug("Total required size: %s", total_space)

```